src/main/java/com/firstspringmvcapp/services/PeopleService.java

```diff
@@ -35,22 +39,21 @@ public List<Person> findAll() {
     }
 
     public Person findOne(int id) {
-        return peopleRepository.findById(id).orElse(null);
+        return peopleRepository.findById(id).orElseThrow(PersonNotFoundException::new);
     }
 
-    public Person findByName(String name) {
-        return peopleRepository.findByName(name).orElse(null);
+    public Person findByEmail(String email) {
+        return peopleRepository.findByEmail(email).orElse(null);
     }
 
-    //@Transactional
-    public List<Person> findFriends(int id) {
+    public Set<Person> findFriends(int id) {
         Person person = peopleRepository.getOne(id);
 
         Hibernate.initialize(person.getFriends());
         Hibernate.initialize(person.getFriendOf());
 
-        List<Person> friendList = person.getFriends();
-        List<Person> friendsOf = person.getFriendOf();
+        Set<Person> friendList = person.getFriends();
+        Set<Person> friendsOf = person.getFriendOf();
 
         friendList.addAll(friendsOf);
```

Comment:
I think that it's not correct to split fiends and friendsOf. Is there any difference? If we are friends it is a two-way communication